### DISCUSSION TITLE
How to use the translation package outside a Symfony application

### DISCUSSION THREAD

**Discussion by robopzet** [symfony/symfony#49962] [Q&A]
In an application without other Symfony packages, I want all text to be configurable by locale. For that I hope I can use the **symfony/translation** package. But the [package documentation](https://symfony.com/doc/current/translation.html) seems to assume the package is always part of a Symfony application.  

Is does not describe, or I've missed it, how to create an implementation of `TranslatorInterface` and make it use my translation files.

Does anyone know how to do that?

**Comment by ghost** [comment]
without FrameworkBundle you have to do all the configuration and instantiation by yourself. I can't find documentation for this either so you have to jump into the source code, you can also inspect how framework bundle does things. There is only small example in README https://github.com/symfony/translation

**Comment by robopzet** [comment]
There's not much documentation in the code, so understanding how to use it is not that easy. And I do not need a lot of included features.

Before diving into this package I'll try the [intl extension](https://www.php.net/manual/en/book.intl.php) first.

**Comment by welcoMattic** [comment]
@robopzet Hi! Couldn't you do the same thing as described in the [README.md](https://github.com/symfony/translation#getting-started) of Translation package?

```php
use Symfony\Component\Translation\Translator;
use Symfony\Component\Translation\Loader\ArrayLoader;

$translator = new Translator('fr_FR');
$translator->addLoader('array', new ArrayLoader());
$translator->addResource('array', [
    'Hello World!' => 'Bonjour !',
], 'fr_FR');

echo $translator->trans('Hello World!'); // outputs « Bonjour ! »
```

Instead of use an `ArrayLoader`, you can use anyone of the [provided loaders](https://github.com/symfony/translation/tree/6.2/Loader) such as the [XliffFileLoader](https://github.com/symfony/translation/blob/6.2/Loader/XliffFileLoader.php).

I guess you don't have to make your own implementation of `TranslatorInterface`.

**Comment by robopzet** [comment]
@welcoMattic Perhaps that would work. But I followed the package documentation on symfony.com. It mentions this after installation with Composer:

>The previous command creates an initial config file where you can define the default locale of the application and the directory where the translation files are located

On my computer no config file was created, so I thought it would not be simple to add to my project.

My solution uses the Intl classes and seems to be good enough. And it's a lot smaller. So I'll keep that..

**Comment by welcoMattic** [comment]
@robopzet this documentation is about installing the package in a Symfony app, not outside. You will not have access to a YAML configuration file, as it's dedicated to FrameworkBundle. In your case, you will have to instantiate and configure manually (basically with PHP, in your own code) how the Translator service will work.
